examples/Example/src/new_api/bottom_sheet/index.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useRef, useState } from 'react';
+import {
+  Dimensions,
+  NativeScrollEvent,
+  NativeSyntheticEvent,
+  StyleSheet,
+  View,
+} from 'react-native';
+import {
+  Gesture,
+  GestureDetector,
+  NativeViewGestureHandler,
+  PanGestureHandlerEventPayload,
+} from 'react-native-gesture-handler';
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+} from 'react-native-reanimated';
+import { LoremIpsum } from '../../../src/common';
+
+const HEADER_HEIGTH = 50;
+const windowHeight = Dimensions.get('window').height;
+const SNAP_POINTS_FROM_TOP = [50, windowHeight * 0.4, windowHeight * 0.8];
+
+const FULLY_OPEN_SNAP_POINT = SNAP_POINTS_FROM_TOP[0];
+const CLOSED_SNAP_POINT = SNAP_POINTS_FROM_TOP[SNAP_POINTS_FROM_TOP.length - 1];
+
+function Example() {
+  const nativeViewRef = useRef(Gesture.Pan());
+  const panGestureRef = useRef(Gesture.Pan());
+  const blockScrollUntilAtTheTopRef = useRef(Gesture.Tap());
+  const [snapPoint, setSnapPoint] = useState(CLOSED_SNAP_POINT);
+  const translationY = useSharedValue(0);
+  const scrollOffset = useSharedValue(0);
+  const bottomSheetTranslateY = useSharedValue(CLOSED_SNAP_POINT);
+
+  const saveScrollOffset = (e: NativeSyntheticEvent<NativeScrollEvent>) => {
+    scrollOffset.value = e.nativeEvent.contentOffset.y;
+  };
+
+  const onPanHandlerUpdate = (e: PanGestureHandlerEventPayload) => {
+    translationY.value = e.translationY - scrollOffset.value;
+  };
+
+  const onHeaderHandlerUpdate = (e: PanGestureHandlerEventPayload) => {
+    translationY.value = e.translationY;
+  };
+
+  const onHeaderHandlerEnd = (e: PanGestureHandlerEventPayload) => {
+    onHandlerEnd(e);
+  };
+
+  const onPanHandlerEnd = (e: PanGestureHandlerEventPayload) => {
+    onHandlerEnd(e);
+  };
```

Comment:
Can't we inline those?

Reply:
Yeah previously logic was different, will update.